src/libsystemd-network/sd-dhcp-client.c

```diff
@@ -282,6 +284,24 @@ int sd_dhcp_client_set_ifindex(sd_dhcp_client *client, int ifindex) {
         return 0;
 }
 
+int sd_dhcp_client_set_ifname(sd_dhcp_client *client, const char *ifname) {
+        assert_return(client, -EINVAL);
+        assert_return(ifname, -EINVAL);
+
+        if (!ifname_valid(ifname))
+                return -EINVAL;
+
+        strcpy(client->ifname, ifname);
+        return 0;
+}
+
+const char *sd_dhcp_client_get_ifname(sd_dhcp_client *client) {
+        if (!client)
+                return NULL;
+
+        return empty_to_null(client->ifname);
+}
```

Comment:
would't it make sense to just fill in ifname lazily and automatically if it wasn't set explicitly? i.e. just calle format_ifname() or so here if ifname is unset, given that we definitely have ifindex around. would be useful for testing things at the very lest